upup/pkg/fi/cloudup/urls.go

```diff
@@ -118,19 +119,20 @@ func SetKopsAssetsLocations(assetsBuilder *assets.AssetBuilder) error {
 }
 
 // NodeUpAsset returns the asset for where nodeup should be downloaded
-func NodeUpAsset(assetsBuilder *assets.AssetBuilder) (*MirroredAsset, error) {
-	// Avoid repeated logging
-	if nodeUpAsset != nil {
+func NodeUpAsset(assetsBuilder *assets.AssetBuilder, arch architectures.Architecture) (*MirroredAsset, error) {
+	if nodeUpAsset == nil {
+		nodeUpAsset = make(map[architectures.Architecture]*MirroredAsset)
+	} else if nodeUpAsset[arch] != nil {
 		// Avoid repeated logging
-		klog.V(8).Infof("Using cached nodeup location: %v", nodeUpAsset.Locations)
-		return nodeUpAsset, nil
+		klog.V(8).Infof("Using cached nodeup location: %v", nodeUpAsset[arch].Locations)
+		return nodeUpAsset[arch], nil
 	}
 	env := os.Getenv("NODEUP_URL")
```

Comment:
We're going to need to change this logic to support different arch urls
```suggestion
	envvar := strings.ToUpper(fmt.Sprintf("NODEUP_URL_%s", arch))
	env := os.Getenv(envvar)
```

and probably also change the messages to `return nil, fmt.Errorf("unable to parse env var %s %q as a url: %v", envvar, env, err)` and 
`klog.Warningf("Using nodeup location from %s env var: %q", envvar, u.String())` below this too

Reply:
Thanks for the feedback @coreypobrien.
I think I will drop completely the requirement for 1.19+. Works pretty well with older versions.
The only supported Docker version will be 19.03.11 (or whatever the default will be in Kops 1.19). Do you see any reason to ad support for older versions?
Adding env vars for multi-arch nodeup seems like a good idea, but should be kept along with the simple one, that will work only on AMD64.

Reply:
Agreed, I don't know of any reason we'd have to do older docker versions. Newest on each OS seems reasonable to me.

On the variables, we need to support supplying both the amd and the arm override (as well as future architectures?) so I thought it would be clearer with `NODEUP_URL_AMD64` and `NODEUP_URL_ARM64`, but as long as we can specify one per architecture the naming doesn't matter much.

Reply:
Perhaps allow `NODEUP_URL_AMD64`, with fallback to `NODEUP_URL` if that doesn't exist.

Reply:
Yes, that is what I meant, but this will probably be the only env var for multi-arch.
For people that want more, there is `KOPS_BASE_URL` and that should be enough.